vehicles/truckstate.cpp

```diff
@@ -20,6 +20,65 @@ void TruckState::updateOdomPositionAndYaw(double drivenDistance, PosType usePosT
     CarState::updateOdomPositionAndYaw(drivenDistance, usePosType);
 }
 
+double TruckState::getCurvatureToPointInVehicleFrame(const QPointF &point)
+{
+    if (mHasTrailer)
+        return getCurvatureWithTrailer(point);
+    else {
+        // calc steering angle (pure pursuit)
+        double distanceSquared = pow(point.x(), 2) + pow(point.y(), 2);
+        double steeringAngleProportional = (2*point.y()) / distanceSquared;
+
+        return -steeringAngleProportional;
+    }
+}
+
+double TruckState::getCurvatureWithTrailer(const QPointF &point)
+{
+    // The point has gone through coordinateTransforms::ENUToVehicleFrame -> vehiclePos is 0,0 with 0 rad yaw
+    // Calculations should be done in vehicle frame
```

Comment:
I think we only need to make some changes to this function and then its ready to merge